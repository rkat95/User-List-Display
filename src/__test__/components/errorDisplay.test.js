import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import ErrorDisplay from "../../components/errorDisplay/errorDisplay";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("Error Display tests", () => {
  it("should render title container", () => {
    render(<ErrorDisplay errorMessage="Error" />);
    const heading = screen.getByTestId("error-container");
    expect(heading).toBeInTheDocument();
  });

  it("should render title div", () => {
    render(<ErrorDisplay errorMessage="Error" />);
    const heading = screen.getByTestId("error-message");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Error");
  });

  it("should match renderer", () => {
    const tree = renderer
      .create(<ErrorDisplay errorMessage="Error" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
