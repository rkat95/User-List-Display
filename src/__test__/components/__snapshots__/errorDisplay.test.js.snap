// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Display tests should match renderer 1`] = `
<div
  className="error-wrapper"
  data-testid="error-container"
>
  <div
    className="error-message"
    data-testid="error-message"
  >
    Error
  </div>
</div>
`;
